### PR TITLE
Posts api

### PR DESCRIPTION
**Pull request by FayeCarter**
We have things

### CODE REVIEW

File: app/controllers/api/v1/posts_controller.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+#  def edit
+#    @post = Post.find(params[:id])
+#  end
+
+  def create
+    @post = Post.create(post_params)
+    if @post.save
+      render json: @post, location: app_v1_post_url(@post) 
+    else
+      render json: @post.errors, status: :unprocessable_entity
```
Comment (hturnbull93) [inline app/controllers/api/v1/posts_controller.rb:19]: Cool json rendering :)

File: app/controllers/api/v1/posts_controller.rb [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    if @post.update(post_params)
+      render json: @posts
+    else
+      render json: @post.errors, status: :unprocessable_entry
+    end
+  end
+
+  def index
+    @posts = Post.all
+    render json: @posts
```
Comment (hturnbull93) [inline app/controllers/api/v1/posts_controller.rb:34]: amazing rendering to json :100:

File: config/application.rb [hunk 3]
```diff
@@ -10,6 +10,8 @@
 module Acebook
   class Application < Rails::Application
     # Initialize configuration defaults for originally generated Rails version.
+    
+    config.api_only = true
```
Comment (hturnbull93) [inline config/application.rb:14]: excellent configuration :1st_place_medal:

File: config/routes.rb [hunk 4]
```diff
@@ -1,12 +1,16 @@
 Rails.application.routes.draw do
+  namespace :api do
+    namespace :v1 do
```
Comment (hturnbull93) [inline config/routes.rb:3]: Long live v1! :100:

File: db/migrate/20200512104515_add_slug_to_posts.rb [hunk 5]
```diff
@@ -0,0 +1,5 @@
+class AddSlugToPosts < ActiveRecord::Migration[5.1]
+  def change
+    add_column :posts, :slug, :string
```
Comment (hturnbull93) [inline db/migrate/20200512104515_add_slug_to_posts.rb:3]: Nice migration!

File: spec/features/api_can_serve_all_posts_spec.rb [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+RSpec.feature "Posts API", type: :feature do
+  scenario "Visiting /api/v1/posts returns all posts" do
+
+    Post.create(message:"Hello!", slug:"HelloWorld")
+    visit '/api/v1/posts'
+    # expect(page).to have_content('"id": 1')
+    expect(page).to have_content('"message":"Hello!"')
+    expect(page).to have_content('"slug":"HelloWorld"')
+  end
+end
```
Comment (hturnbull93) [inline spec/features/api_can_serve_all_posts_spec.rb:14]: Excellent test! is a post having a slug the functionality we want what does a slug do?

File: spec/features/user_can_edit_post_spec.rb [hunk 7]
```diff
@@ -1,17 +1,17 @@
...
+#     fill_in "Message", with: "Hello there!"
+#     click_button "Submit"
+#     expect(page).to have_content("Hello there!")
+#     click_on 'Edit'
+#     fill_in "Message", with: "General Kenobi!"
+#     click_button "Submit"
+#     expect(page).to_not have_content("Hello there!")
+#     expect(page).to have_content("General Kenobi!")
+#   end
+# end
```
Comment (hturnbull93) [inline spec/features/user_can_edit_post_spec.rb:17]: F

File: spec/features/user_can_login_spec.rb [hunk 8]
```diff
@@ -1,15 +1,15 @@
...
+#     fill_in "user[email]", with: "samgangi@lotr.com"
+#     fill_in "user[password]", with: "theshire"
+#     click_button "Submit"
 
-    expect(page).to have_content("Welcome Sam")
-  end
-end
+#     expect(page).to have_content("Welcome Sam")
+#   end
+# end
```
Comment (hturnbull93) [inline spec/features/user_can_login_spec.rb:15]: F

File: spec/features/user_can_singup_spec.rb [hunk 9]
```diff
@@ -1,30 +1,30 @@
...
+#     fill_in "user[password]", with: "password"
+#     click_button "Submit"
     
 
-    expect(page).to have_content("That email is already in use")
-  end
-end
+#     expect(page).to have_content("That email is already in use")
+#   end
+# end
```
Comment (hturnbull93) [inline spec/features/user_can_singup_spec.rb:30]: F

File: spec/features/user_can_submit_posts_spec.rb [hunk 10]
```diff
@@ -1,16 +1,16 @@
...
-  end
-end
+#     visit "/posts"
+#     click_link "New post"
+#     fill_in "Message", with: "Hello, world!"
+#     click_button "Submit"
+#     expect(page).to have_content("Hello, world!")
+#     expect(page).to have_content(formatted_time)
+#   end
+# end
```
Comment (hturnbull93) [inline spec/features/user_can_submit_posts_spec.rb:16]: F

File: spec/features/user_can_view_posts_spec.rb [hunk 11]
```diff
@@ -1,15 +1,15 @@
...
+#     click_link "New post"
+#     fill_in "Message", with: "first post!"
+#     click_button "Submit"
+#     visit "/posts"
+#     click_link "New post"
+#     fill_in "Message", with: "second post"
+#     click_button "Submit"
+#     expect(first(".post")).to have_content("second post")
+#   end
+# end
```
Comment (hturnbull93) [inline spec/features/user_can_view_posts_spec.rb:15]: F (these are jokes for Ed btw)

File: spec/features/user_cannot_sign_in_with_invalid_credentials_spec.rb [hunk 12]
```diff
@@ -1,26 +1,26 @@
...
+#     fill_in "user[email]", with: "samgangi@lotr.com"
+#     fill_in "user[password]", with: "sauron"
+#     click_button "Submit"
 
-    expect(page).to have_content("Incorrect email/password")
-  end
-end
+#     expect(page).to have_content("Incorrect email/password")
+#   end
+# end
```
Comment (hturnbull93) [inline spec/features/user_cannot_sign_in_with_invalid_credentials_spec.rb:26]: F

**Review by hturnbull93 — APPROVED** [review]

File: app/controllers/api/v1/posts_controller.rb [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+      render json: @post.errors, status: :unprocessable_entity
+    end
+  end
+
+  def update
+    @post = Post.find(params[:id])
+    if @post.update(post_params)
+      render json: @posts
+    else
+      render json: @post.errors, status: :unprocessable_entry
```
Comment (tommyrharper) [inline app/controllers/api/v1/posts_controller.rb:28]: json rendering makes sense and is easily readable

File: app/controllers/api/v1/posts_controller.rb [hunk 14]
```diff
@@ -0,0 +1,54 @@
...
+    end
+  end
+
+  def index
+    @posts = Post.all
+    render json: @posts
+  end
+
+  def show
+    render json: @post
```
Comment (tommyrharper) [inline app/controllers/api/v1/posts_controller.rb:38]: fantastic simple code

File: config/routes.rb [hunk 15]
```diff
@@ -1,12 +1,16 @@
...
+      resources :posts
+      resources :sessions, only:[:new, :create, :destroy]
```
Comment (tommyrharper) [inline config/routes.rb:5]: Good code

File: db/schema.rb [hunk 16]
```diff
@@ -10,7 +10,7 @@
 #
 # It's strongly recommended that you check this file into your version control system.
 
-ActiveRecord::Schema.define(version: 20200507101056) do
+ActiveRecord::Schema.define(version: 20200512104515) do
```
Comment (tommyrharper) [inline db/schema.rb:13]: nice new schema

File: spec/features/api_can_serve_all_posts_spec.rb [hunk 17]
```diff
@@ -0,0 +1,14 @@
+# localhost:3000/api/v1/posts
+
+require 'rails_helper'
+
+RSpec.feature "Posts API", type: :feature do
+  scenario "Visiting /api/v1/posts returns all posts" do
```
Comment (tommyrharper) [inline spec/features/api_can_serve_all_posts_spec.rb:6]: good testing

**Pull request merged**